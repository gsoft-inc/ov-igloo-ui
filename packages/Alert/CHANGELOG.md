# @igloo-ui/alert

## 0.1.12

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2
- Updated dependencies [0640409]
- Updated dependencies [0640409]
  - @igloo-ui/button@0.3.6
  - @igloo-ui/icon-button@0.2.12

## 0.1.11

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme
- Updated dependencies [d971143]
  - @igloo-ui/button@0.3.5
  - @igloo-ui/icon-button@0.2.11

## 0.1.10

### Patch Changes

- d846589: Updated alert scss to match design

## 0.1.9

### Patch Changes

- 1a3322f: Upgrade igloo icons package from `1.1.0` to `1.3.0`

## 0.1.8

### Patch Changes

- 38a4994: Update dependency on @igloo-ui/icons and other @igloo libs
- Updated dependencies [4de58ec]
- Updated dependencies [38a4994]
  - @igloo-ui/button@0.3.0
  - @igloo-ui/icon-button@0.2.9

## 0.1.7

### Patch Changes

- 0a86f13: Update content of README
- Updated dependencies [0a86f13]
  - @igloo-ui/button@0.2.2
  - @igloo-ui/icon-button@0.2.8

## 0.1.6

### Patch Changes

- Updated dependencies [f019f88]
  - @igloo-ui/button@0.2.0
  - @igloo-ui/icon-button@0.2.7

## 0.1.5

### Patch Changes

- cb767c6: align icon on top of component

## 0.1.4

### Patch Changes

- b8b8912: fix: remove dependence of tokens.css

## 0.1.3

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files
- Updated dependencies [8c8686e]
  - @igloo-ui/button@0.1.11
  - @igloo-ui/icon-button@0.2.6

## 0.1.2

### Patch Changes

- 0c48abc: update UI

## 0.1.1

### Patch Changes

- 4a256e3: IE11 compatibility
- Updated dependencies [4a256e3]
  - @igloo-ui/button@0.1.9
  - @igloo-ui/icon-button@0.2.3

## 0.1.0

### Minor Changes

- 489b353: Initial release

### Patch Changes

- Updated dependencies [0d8e80e]
  - @igloo-ui/icon-button@0.2.1
