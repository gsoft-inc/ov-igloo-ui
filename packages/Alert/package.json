{
  "name": "@igloo-ui/alert",
  "version": "0.6.5",
  "main": "dist/Alert.js",
  "module": "dist/Alert.js",
  "types": "dist/Alert.d.ts",
  "styles": "dist/alert.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "alexandre.beaudoin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/button": "^0.7.7",
    "@igloo-ui/icon-button": "^1.1.7",
    "@igloo-ui/icons": "^1.12.2",
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.1.3",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@igloo-ui/tooltip": "^3.4.4",
    "@igloo-ui/hyperlink": "^0.3.4"
  }
}
