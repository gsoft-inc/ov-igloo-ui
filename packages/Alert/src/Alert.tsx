import * as React from "react";
import classNames from "classnames";

import Button, { type Appearance as ButtonAppearance } from "@igloo-ui/button";
import IconButton from "@igloo-ui/icon-button";
import { DismissIcon } from "@hopper-ui/icons-react16";
import {
    InfoIcon,
    CrownIcon,
    SuccessIcon,
    WarningIcon
} from "./svgs";
import {
    TadaIcon as LegacyTadaIcon,
    InfoIcon as LegacyInfoIcon,
    CrownIcon as LegacyCrownIcon,
    SuccessIcon as LegacySuccessIcon,
    WarningIcon as LegacyWarningIcon
} from "./legacy/svgs";
import { useLocalizedStringFormatter } from "@igloo-ui/provider";
import intlMessages from "./intl";

import "./alert.scss";

export type Type = "announcement" | "info" | "premium" | "success" | "warning";

export type Appearance = "card" | "inline" | "horizontal";

export interface AlertButton {
    label: string;
    onClick: () => void;
}

export interface AlertProps extends Omit<React.ComponentProps<"div">, "title"> {
    /** The title of the alert */
    title?: React.ReactNode;
    /** The content to display inside the Alert */
    message?: React.ReactNode;
    /** Additional text to display at the top right of the Alert */
    metadata?: React.ReactNode;
    /** Change the type of the Alert */
    type: Type;
    /** Change the Alert appearance */
    appearance?: Appearance;
    /** Add a specific class to the Alert */
    className?: string;
    /** Set if the Alert can be closed by the user */
    closable?: boolean;
    /** Custom icon for alert (Use 'null' to have no icon) */
    icon?: React.ReactElement | null;
    /** Action on Alert close button click */
    onClose?: () => void;
    /** Alert button */
    button?: AlertButton;
    /** Add a data-test tag for automated tests */
    dataTest?: string;
}

const getBrand = (): string => {
    return document.documentElement.getAttribute("data-brand") ?? "igloo";
};

const renderIcon = (
    style: Appearance,
    hasButton: boolean,
    type: Type,
    icon?: React.ReactElement,
    isWorkleap?: boolean
): JSX.Element => {
    const classes = classNames("ids-alert__icon", `ids-alert__icon--${style}`, {
        "ids-alert__icon--small-top": hasButton,
        "ids-alert__icon--medium-centered": !hasButton
    });

    return (
        <div className={classes}>
            {icon || (
                <>
                    {type === "announcement" && (isWorkleap ? <InfoIcon /> : <LegacyTadaIcon />)}
                    {type === "info" && (isWorkleap ? <InfoIcon /> : <LegacyInfoIcon />)}
                    {type === "premium" && (isWorkleap ? <CrownIcon /> : <LegacyCrownIcon />)}
                    {type === "success" && (isWorkleap ? <SuccessIcon /> : <LegacySuccessIcon />)}
                    {type === "warning" && (isWorkleap ? <WarningIcon /> : <LegacyWarningIcon />)}
                </>
            )}
        </div>
    );
};

const renderDismissButton = (
    setShow: (show: boolean) => void,
    onDismissClick?: () => void,
    ariaLabel?: string
): JSX.Element => {
    const action = (): void => {
        if (onDismissClick) {
            onDismissClick();
        }

        setShow(false);
    };

    return (
        <IconButton
            appearance={{ type: "ghost", variant: "secondary" }}
            className="ids-alert__dismiss-btn"
            type="button"
            size="xsmall"
            icon={<DismissIcon size="sm" />}
            onClick={action}
            aria-label={ariaLabel}
        />
    );
};

const renderAlertActionButton = (
    style: Appearance,
    button?: AlertButton,
    isWorkleap?: boolean
): JSX.Element => {
    if (button == null || button.onClick == null || button.label == null) {
        return <></>;
    }

    const ghostAppearance = isWorkleap ? 
        { type: "ghost", variant: "secondary" } as ButtonAppearance : 
        "ghost";

    return (
        <Button
            appearance={style === "horizontal" ? ghostAppearance : "secondary"}
            size="small"
            onClick={button.onClick}
            className="ids-alert__action-btn"
        >
            {button.label}
        </Button>
    );
};

const Alert: React.FunctionComponent<AlertProps> = ({
    title,
    message,
    metadata,
    type,
    appearance = "card",
    className,
    closable = true,
    icon,
    onClose,
    button,
    dataTest,
    ...rest
}: AlertProps) => {
    const stringFormatter = useLocalizedStringFormatter(intlMessages);
    const classes = classNames(
        "ids-alert",
        `ids-alert--${appearance}`,
        `ids-alert--${type}`,
        className
    );

    const parentElement = React.useRef<HTMLDivElement>(null);
    const [show, setShow] = React.useState(true);
    const hasButton = button !== undefined;
    const isHorizontal = appearance === "horizontal";
    const canBeClosed = closable && !isHorizontal;
    const isWorkleap = getBrand() === "workleap";

    if (show) {
        return (
            <div
                className={classes}
                ref={parentElement}
                data-test={dataTest}
                {...rest}
            >
                {icon !== null &&
          !isHorizontal &&
          renderIcon(appearance, hasButton, type, icon, isWorkleap)}

                <div className="ids-alert__body">
                    <div className="ids-alert__header">
                        <p className="ids-alert__title">{title}</p>
                        <p className="ids-alert__metadata">{metadata}</p>
                    </div>
                    {!isHorizontal && <div className="ids-alert__content">{message}</div>}
                    {hasButton && renderAlertActionButton(appearance, button, isWorkleap)}
                </div>

                {canBeClosed && renderDismissButton(setShow, onClose, stringFormatter.format("close"))}
            </div>
        );
    }

    return null;
};

export default Alert;
