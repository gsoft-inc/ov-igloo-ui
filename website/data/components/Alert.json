[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Alert/src/Alert.tsx",
      "description": "",
      "displayName": "Alert",
      "methods": [],
      "props": {
        "title": {
          "defaultValue": null,
          "description": "The title of the alert",
          "name": "title",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "message": {
          "defaultValue": null,
          "description": "The content to display inside the Alert",
          "name": "message",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "type": {
          "defaultValue": null,
          "description": "Change the type of the Alert",
          "name": "type",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": true,
          "type": { "name": "Type" }
        },
        "appearance": {
          "defaultValue": null,
          "description": "Change the Alert appearance",
          "name": "appearance",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "Appearance" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the Alert",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "closable": {
          "defaultValue": null,
          "description": "Set if the Alert can be closed by the user",
          "name": "closable",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "onClose": {
          "defaultValue": null,
          "description": "Action on Alert close button click",
          "name": "onClose",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "() => void" }
        },
        "button": {
          "defaultValue": null,
          "description": "Alert button",
          "name": "button",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "AlertButton" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Alert/src/Alert.tsx",
            "name": "AlertProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Alert/src/Alert.tsx",
              "name": "AlertProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
