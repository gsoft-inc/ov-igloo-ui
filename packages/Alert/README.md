# @igloo-ui/alert

Welcome to my `Alert` component.

## Installation

```sh
npm i @igloo-ui/alert
# or with yarn
yarn add @igloo-ui/alert
```

Then to use the component in your code just import it!

```js
import Alert from '@igloo-ui/alert';
// and with css-modules
import '@igloo-ui/alert/dist/alert.css';
```
