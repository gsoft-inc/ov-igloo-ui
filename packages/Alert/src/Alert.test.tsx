/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Alert, { Type, AlertProps, Style } from './Alert';

const setup = (props: AlertProps) => {
  return shallow(<Alert {...props}>Hello world</Alert>);
};

describe('Alert', () => {
  const component = (type: Type) => setup({ type });

  const expectToBeOfType = (alert: ShallowWrapper, type: Type) => {
    const findAlert = alert.find('.ids-alert');

    const findAnnouncement = alert.find('.ids-alert--announcement');
    const findInfo = alert.find('.ids-alert--info');
    const findPremium = alert.find('.ids-alert--premium');
    const findPromo = alert.find('.ids-alert--promo');
    const findSuccess = alert.find('.ids-alert--success');
    const findWarning = alert.find('.ids-alert--warning');

    expect(findAlert.length).toBe(1);

    expect(findAnnouncement.length === 1).toBe(type === 'announcement');
    expect(findInfo.length === 1).toBe(type === 'info');
    expect(findPremium.length === 1).toBe(type === 'premium');
    expect(findPromo.length === 1).toBe(type === 'promo');
    expect(findSuccess.length === 1).toBe(type === 'success');
    expect(findWarning.length === 1).toBe(type === 'warning');
  };

  const expectTpBeOfStyle = (alert: ShallowWrapper, style: Style) => {
    const findCard = alert.find('.ids-alert--card');
    const findInline = alert.find('.ids-alert--inline');
    const findHorizontal = alert.find('.ids-alert--horizontal');

    expect(findCard.length === 1).toBe(style === 'card');
    expect(findInline.length === 1).toBe(style === 'inline');
    expect(findHorizontal.length === 1).toBe(style === 'horizontal');
  };

  test('It should render without errors', () => {
    const wrapper = component('none').find('.ids-alert');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component('none')).toMatchSnapshot();
  });

  test('It should render an alert with type "announcement"', () => {
    const alert = component('announcement');

    expectToBeOfType(alert, 'announcement');
  });

  test('It should render an alert with type "info"', () => {
    const alert = component('info');

    expectToBeOfType(alert, 'info');
  });

  test('It should render an alert with type "premium"', () => {
    const alert = component('premium');

    expectToBeOfType(alert, 'premium');
  });

  test('It should render an alert with type "promo"', () => {
    const alert = component('promo');

    expectToBeOfType(alert, 'promo');
  });

  test('It should render an alert with type "success"', () => {
    const alert = component('success');

    expectToBeOfType(alert, 'success');
  });

  test('It should render an alert with type "warning"', () => {
    const alert = component('warning');

    expectToBeOfType(alert, 'warning');
  });

  test('It should render an alert with type "none"', () => {
    const alert = component('none');

    expectToBeOfType(alert, 'none');
  });

  test('It should render by default an alert with a dismiss button', () => {
    const alert = setup({ type: 'none' });
    const findDismissButton = alert.find('.ids-alert__dismiss-button');

    expect(findDismissButton.length).toBe(1);
  });

  test('It should render an alert with a dismiss button', () => {
    const alert = setup({ type: 'none', isDismissible: true });
    const findDismissButton = alert.find('.ids-alert__dismiss-button');

    expect(findDismissButton.length).toBe(1);
  });

  test('It should render an alert without a dismiss button', () => {
    const alert = setup({ type: 'none', isDismissible: false });
    const findDismissButton = alert.find('.ids-alert__dismiss-button');

    expect(findDismissButton.length).toBe(0);
  });

  test('It should render by default a Card style alert', () => {
    const alert = setup({ type: 'none' });

    expectTpBeOfStyle(alert, 'card');
  });

  test('It should render a Card style alert', () => {
    const alert = setup({ type: 'none', alertStyle: 'card' });

    expectTpBeOfStyle(alert, 'card');
  });

  test('It should render a Inline style alert', () => {
    const alert = setup({ type: 'none', alertStyle: 'inline' });

    expectTpBeOfStyle(alert, 'inline');
  });

  test('It should render a Horizontal style alert', () => {
    const alert = setup({ type: 'none', alertStyle: 'horizontal' });

    expectTpBeOfStyle(alert, 'horizontal');
  });

  test('It should render an Alert without action button', () => {
    const alert = setup({ type: 'none', alertStyle: 'card' });
    const findActionButton = alert.find('Button');

    expect(findActionButton.length).toBe(0);
  });

  test('It should render an Alert with action button', () => {
    const buttonText = 'Supercalifragilisticexpialidocious';
    const alert = setup({
      type: 'none',
      alertStyle: 'card',
      alertActionText: buttonText,
      onAlertActionClick: () => {
        console.log(buttonText);
      },
    });
    const findActionButton = alert.find('Button');

    expect(findActionButton.length).toBe(1);
    expect(findActionButton.html()).toContain(buttonText);
  });

  test('It should render a Card Alert with an icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'card',
    });
    const findIcon = alert.find('.ids-alert__icon');

    expect(findIcon.length).toBe(1);
  });

  test('It should render a Inline Alert with an icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'inline',
    });
    const findIcon = alert.find('.ids-alert__icon');

    expect(findIcon.length).toBe(1);
  });

  test('It should render a Horizontal Alert without an icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'horizontal',
    });
    const findIcon = alert.find('.ids-alert__icon');

    expect(findIcon.length).toBe(0);
  });

  test('It should render an Alert by default with a medium icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'card',
    });
    const findIcon = alert.find('.ids-alert__icon--medium-centered');

    expect(findIcon.length).toBe(1);
  });

  test('It should render an Alert with a medium icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'card',
      iconStyle: 'medium-centered',
    });
    const findIcon = alert.find('.ids-alert__icon--medium-centered');

    expect(findIcon.length).toBe(1);
  });

  test('It should render an Alert with a small icon', () => {
    const alert = setup({
      type: 'success',
      alertStyle: 'card',
      iconStyle: 'small-top',
    });
    const findIcon = alert.find('.ids-alert__icon--small-top');

    expect(findIcon.length).toBe(1);
  });
});
