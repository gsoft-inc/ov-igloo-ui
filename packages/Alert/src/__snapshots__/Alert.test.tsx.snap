// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert It should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-alert ids-alert--card ids-alert--info"
  >
    <div
      class="ids-alert__icon ids-alert__icon--card ids-alert__icon--medium-centered"
    >
      <svg
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="16"
          cy="16"
          fill="#B2DDF2"
          r="16"
        />
        <path
          d="M16.6158 7.39951C17.34 8.52136 17.0484 9.99799 15.9645 10.6977C14.8807 11.3973 13.4149 11.0551 12.6908 9.93323C11.9666 8.81138 12.2582 7.33475 13.342 6.63509C14.4259 5.93542 15.8916 6.27767 16.6158 7.39951Z"
          fill="white"
        />
        <path
          d="M17.5039 13.6C16.8639 13.6 14.5706 15.2 13.5039 16C9.50391 19.2 14.7039 17.2 15.1039 18.4C15.4239 19.36 15.7706 24.1333 15.9039 26.4C15.9274 26.8 20.2449 26.517 20.8 26C22.1927 24.703 19.4993 24.2993 18.8 23.6C18.4 23.2 18.3039 13.6 17.5039 13.6Z"
          fill="white"
        />
        <path
          clip-rule="evenodd"
          d="M14.4 7.2C13.5163 7.2 12.8 7.91634 12.8 8.8C12.8 9.68365 13.5163 10.4 14.4 10.4C15.2837 10.4 16 9.68365 16 8.8C16 7.91634 15.2837 7.2 14.4 7.2ZM11.2 8.8C11.2 7.03269 12.6327 5.6 14.4 5.6C16.1673 5.6 17.6 7.03269 17.6 8.8C17.6 10.5673 16.1673 12 14.4 12C12.6327 12 11.2 10.5673 11.2 8.8Z"
          fill="#233043"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M16.1657 13.8343C16.4781 14.1467 16.4781 14.6533 16.1657 14.9657C15.7262 15.4052 15.1488 16.0779 14.5585 16.7694C14.5313 16.8013 14.504 16.8333 14.4767 16.8653C14.3774 16.9816 14.2777 17.0984 14.1787 17.214C14.3856 17.2056 14.5962 17.2 14.8 17.2C15.1982 17.2 15.5357 17.4927 15.592 17.8869L16.6939 25.6H19.6687L18.2344 24.1657C18.1038 24.0351 18.0224 23.8634 18.004 23.6796L17.204 15.6796C17.16 15.24 17.4808 14.8479 17.9204 14.804C18.3601 14.76 18.7521 15.0808 18.7961 15.5204L19.5677 23.2362L22.1657 25.8343C22.3945 26.0631 22.463 26.4072 22.3391 26.7061C22.2153 27.0051 21.9236 27.2 21.6 27.2H16C15.6019 27.2 15.2644 26.9073 15.2081 26.5131L14.1088 18.8184C13.9024 18.8281 13.6984 18.8404 13.5033 18.8522C13.3696 18.8603 13.24 18.8681 13.1167 18.8747C12.889 18.8868 12.642 18.8971 12.459 18.8829C12.4091 18.879 12.3448 18.872 12.2763 18.8574C12.2223 18.8458 12.0937 18.8154 11.9611 18.7295C11.8112 18.6325 11.5593 18.3884 11.5818 17.9884C11.5993 17.6772 11.7818 17.4869 11.8344 17.4343C12.196 17.0727 12.7175 16.4617 13.2927 15.7878C13.309 15.7688 13.3253 15.7497 13.3416 15.7306C13.9179 15.0554 14.5405 14.3281 15.0344 13.8343C15.3468 13.5219 15.8533 13.5219 16.1657 13.8343Z"
          fill="#233043"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="ids-alert__body"
    >
      <div
        class="ids-alert__header"
      >
        <p
          class="ids-alert__title"
        />
        <p
          class="ids-alert__metadata"
        />
      </div>
      <div
        class="ids-alert__content"
      />
    </div>
    <button
      class="ids-btn ids-icon-btn ids-icon-btn--xsmall ids-alert__dismiss-btn ids-btn--ghost-secondary"
      title=""
      type="button"
    >
      <span
        class=""
      >
        <svg
          aria-hidden="true"
          class="Icon-module__hop-icon___9H6Vf"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M12.947 4.114a.75.75 0 1 0-1.061-1.06L8 6.938 4.114 3.053a.75.75 0 0 0-1.06 1.061L6.938 8l-3.886 3.886a.75.75 0 0 0 1.061 1.06L8 9.062l3.886 3.886a.75.75 0 0 0 1.06-1.061L9.062 8z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
